packages/components/table/src/store/watcher.ts

```diff
@@ -46,6 +46,7 @@ const doFlattenColumns = (columns) => {
 
 function useWatcher<T>() {
   const instance = getCurrentInstance() as Table<T>
+  const { size: tableSize } = toRefs(instance.props)
```

Comment:
I think we should use `instance.$props` instead of `instance.$proxy.$props`(this one is actually reactive)

Reply:
cc @cokemine (

Reply:
It seems that the instance object obtained by getCurrentInstance Hook does not have $props or $proxy properties, we can only get the props through instance.props or instance.ctx(proxy).$props (still reactive)

Reply:
It was marked as optional because you can define a component without props so that would be accurate. For us we know that there must be this $props object you can use `!.` to bypass the type checking.

Reply:
But actually it is always `undefined` when I try to output it

Reply:
that's weird cc @sxzz can you have a look on this? Until we have found the root cause let's stick with your way.

Reply:
It's `instance.proxy.$props`. /cc @JeremyWuuuuu @cokemine 
![image](https://user-images.githubusercontent.com/6481596/140039855-3431079c-fc4d-485e-ba94-14184f7a37ec.png)

Reply:
@cokemine Emmm, we just use instance.proxy.$props
```suggestion
  const { size: tableSize } = toRefs(instance.proxy.$props)
```

Reply:
What's the different between using `instance.props` and `instance.proxy.$props`.
I only know that `instance.ctx` can be only used in dev mode.
BTW, it seems like `instance.proxy.$props` will may lose types

Reply:
Doesn't matter, since `$props` is defined by Vue so we can simply cast it.